## Useful Resources for Continuous Learning

- [Clinical Genomics Guide App](https://www.genomicsapps.org/)
- [EuroGMS](https://www.eurogems.org/): a guide to approximately 110 selected genomic online educational sources

## Common Databases

| S. No | Database                     | Description                                                                 |
|-------|------------------------------|-----------------------------------------------------------------------------|
| 1     | NCBI GenBank                 | Repository for genetic sequences                                           |
| 2     | RefSeq                       | Curated collection of sequences, functional annotations                    |
| 3     | PubMed                       | Database of biomedical literature                                          |
| 4     | OMIM                         | Catalog of human genes and genetic disorders                               |
| 5     | dbSNP                        | Database of single nucleotide polymorphisms                                |
| 6     | EST                          | Expressed Sequence Tag database                                            |
| 7     | dbEST                        | Database of Expressed Sequence Tags                                        |
| 8     | WGS                          | Whole Genome Shotgun Sequence database                                     |
| 9     | HTG                          | High Throughput Genome Sequence database                                   |
| 10    | EMBL                         | European Molecular Biology Laboratory sequence database                    |
| 11    | Array Express                | Archive of functional genomics data                                        |
| 12    | EBI Metagenomics             | Database for metagenomics studies                                          |
| 13    | Ensembl                      | Genome browser for vertebrate genomes                                      |
| 14    | European Nucleotide Archive  | Archive for all European nucleotide sequence data                          |
| 15    | Immuno Polymorphism Database | Database of human immunoglobulin variable genes                            |
| 16    | Pfam                         | Database of protein families and domains                                   |
| 17    | NetPDB                       | Network of protein 3D structure databases                                  |
| 18    | STRING                       | Database of known and predicted protein-protein interactions               |
| 19    | UniProt                      | Comprehensive resource for protein sequence and annotation data            |
| 20    | PRIDE                        | Comprises a comprehensive suite of libraries and command-line tools tailored for computational proteomics |

| Tools                 | Description                                                                                 |
|-----------------------|---------------------------------------------------------------------------------------------|
| KEGG Database         | Database for understanding high-level functions and utilities of biological systems         |
| KEGG PATHWAY          | Collection of manually drawn pathway maps representing molecular interactions and reaction networks |
| DDBJ                  | DNA Data Bank of Japan, a nucleotide sequence database                                      |
| GenomeReviews         | Database containing reviews on complete genomes                                             |
| UniProt               | Compendium of protein sequences and functional information                                  |
| UniProtKB             | Protein sequence and functional information resource                                        |
| UniProt/SwissProt     | Part of the UniProt database, containing manually annotated and reviewed protein sequences   |

## Computational Tools

### A. Similarity-based method

| Tool                 | Description                                                                                                                                  | Algorithm                                | Type of Data                              | Use-case Addressed                                                                 | Programming Language | Advantages                                                                                                                                               | Disadvantages                                                                                                                                         | Challenges                                             | References                               |
|----------------------|----------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------|-------------------------------------------|------------------------------------------------------------------------------------|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------|-----------------------------------------|
| [NEMO](https://acgt.cs.tau.ac.il/nemo/)               | Network Module Enrichment Analysis, used for identifying enriched modules in biological networks based on omics data.                        | Network-based clustering                 | Genomics, Transcriptomics, Proteomics, Metabolomics | Identifying enriched modules and pathways in biological networks                       | Proprietary          | Identifies relevant modules and pathways, and integrates network information.                                       | Reliance on predefined networks, potential for biased interpretations.                                        | Keeping network databases updated, and handling data quality. | Rappaport and Shamir, 2019             |
| [PinPLUS](https://cran.r-project.org/web/packages/PINSPlus/index.html)            | Pinplus is used for pinpointing enriched modules and key interactions within biological networks, leveraging omics data to highlight significant connections and pathways. | Network-based analysis and module enrichment | Genomics, Transcriptomics, Proteomics, Metabolomics | To identify and characterize enriched modules and interactions within biological networks, providing insights into relevant pathways and functional relationships. | Proprietary          | Identifies significant modules and interactions. Integrates diverse omics data for comprehensive network analysis. Provides detailed insights into functional relationships within networks. | May rely on predefined network models, potentially influencing results. Risk of biased interpretations based on network structure and data quality. | Keeping network databases current and relevant. Ensuring data quality and accuracy for meaningful results. | NA                                       |

### B. Network-based method

| Tool                 | Description                                                                                                                                  | Algorithm                                | Type of Data                              | Use-case Addressed                                                                 | Programming Language | Advantages                                                                                                                                               | Disadvantages                                                                                                                                         | Challenges                                             | References                               |
|----------------------|----------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------|-------------------------------------------|------------------------------------------------------------------------------------|----------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------|-----------------------------------------|
| [Steiner](https://cran.r-project.org/src/contrib/Archive/SteinerNet/)            | A tool for network-based analysis to identify key regulatory elements and pathways from omics data.                                           | Steiner tree algorithm                   | Genomics, Transcriptomics, Proteomics, Metabolomics | Identifying key regulators and pathways from omics data                               | Python               | Incorporates network information, and identifies important regulatory components.                                      | Requires expertise in network analysis, computational resources for large datasets.                                 | Handling large-scale networks, interpreting complex regulatory interactions.        | Tuncbag et al., 2012                   |
| [OncoIMPACT](https://github.com/CSBS/OncoIMPACT)          | A tool used for integrative analysis of cancer omics data to identify potential therapeutic targets and biomarkers.                           | Network-based integration, machine learning models | Genomics, Transcriptomics, Proteomics, Metabolomics | Identifying therapeutic targets and biomarkers from Cancer Omics data                 | Proprietary          | Integrates multi-omics data and identifies potential therapeutic targets and biomarkers.                              | Proprietary implementation, potential for biased interpretations.                                                       | Handling data quality issues, and interpreting complex models.                       | Bertrand et al., 2015                  |
| [WGCNA](https://labs.genetics.ucla.edu/horvath/CoexpressionNetwork/Rpackages/WGCNA/)               | Tool for analyzing high-dimensional gene expression data to identify clusters of co-expressed genes and their correlation with phenotypic traits. | Hierarchical clustering                   | Genomics                                   | Identification of candidate biomarkers or therapeutic targets                         | R                    | Identifies co-expression modules, useful for understanding gene networks.                                             | Sensitivity to parameter choices, potential for spurious correlations.                                                 | Handling large-scale datasets, interpreting module relevance biologically.           | Langfelder and Horvath, 2008           |
| [Omics Integrator](https://fraenkel.mit.edu/omicsintegrator)    | A tool for integrating and analyzing multi-omics data to identify biological insights and associations.                                       | Network-based integration                 | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating multi-omics data to identify biological insights and associations         | R                    | Integrates diverse omics data types and incorporates network information.                                             | Keeping network databases updated, and handling large networks.                                                          | Handling and interpreting complex multi-omics datasets.                              | Fraenkel et al., NA                     |
| Weighted Multiplex Networks | A network-based approach for integrating and analyzing multi-omics data with weighted relationships.                                       | Weighted multiplex network construction   | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating and analyzing multi-omics data with weighted relationships                | Python               | Captures complex relationships and incorporates edge weights.                                                          | Potential for increased computational complexity.                                                                       | Handling large networks, interpreting biological relevance.                          | Menchetti et al., 2014                 |

### C. Correlation-based method

| Tool                        | Description                                                                                                                                               | Algorithm                     | Type of Data                              | Use-case Addressed                                                            | Programming Language | Advantages                                                                                                                                 | Disadvantages                                                                                                                         | Challenges                                               | References                         | URL Link                                                    |
|-----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------|-------------------------------------------|--------------------------------------------------------------------------------|----------------------|-------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------|-----------------------------------|------------------------------------------------------------|
| [CONEXIC](https://www.c2b2.columbia.edu/danapeerlab/html/conexic.html)                     | A method used for integrative analysis and visualization of multi-omics data to uncover biological networks and pathways.                                  | Integration of multi-omics data, network reconstruction | Genomics, Transcriptomics, Proteomics, Metabolomics | Uncovering biological networks and pathways from integrated multi-omics data                     | Java                 | Integrates diverse omics data types and identifies key biological networks and drivers.                                                | Potential for biased interpretations, handling noise and missing data.                                                               | Ensuring network robustness, interpreting complex networks biologically. | Akavia et al., 2010               |
| [MapMan](https://mapman.gabipd.org)                      | The tool is used for functional annotation, visualization, and analysis of omics data, particularly in the context of metabolic pathways.                   | HMM                           | Genomics                                   | Enrichment analysis of plant species                                           | Java                 | Effective visualization and analysis of large datasets in plant genomics.                                                             | Limited application outside of plant genomics, may require domain-specific knowledge.                                                | Evolving new plant genomics data and techniques.                         | Schwacke et al., 2019             |
| [CNAnet](http://csbi.ltdk.helsinki.fi/CNAnet/)                      | Tool for Copy Number Alteration Analysis with Metabolomics data, used for integrating copy number variation data with metabolomics data.                   | Permutation tests, correlation analysis | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating copy number variation data with metabolomics data                                   | R                    | Integrates genomic and metabolomics data and accounts for multiple tests.                                                             | Limited to specific data types, potential for spurious correlations.                                                                  | Handling high-dimensional data, and interpreting correlations biologically. | Louhimo and Hautaniemi, 2011      |
| [PhenoLink](http://bamics2.cmbi.ru.nl/websoftware/phenolink/)                   | The tool used for phenotype-based omics data integration and analysis to uncover associations between phenotypic traits and molecular profiles.             | Correlation-based integration  | Genomics                                   | Linking phenotypes to omics data for bacteria                                   | R                    | Integrates phenotypic and omics data, useful for bacteria studies.                                                                    | Limited to specific data types, potential for spurious correlations.                                                                  | Handling high-dimensional omics data, and interpreting correlations biologically. | Bayjanov et al., 2012             |
| [Paintomics](http://www.paintomics.org/)                 | A tool for visualizing and analyzing omics data in the context of biological pathways.                                                                     | NA                            | Genomics, Transcriptomics, Proteomics, Metabolomics | Regulatory function of B-cell differentiation                                  | R, JavaScript        | Combines omics data with pathway information, and visualization aids.                                                                | Reliance on predefined pathways, potential for oversimplification.                                                                    | Ensuring pathway relevance, integrating dynamic pathway data.             | Liu et al., 2022                  |
| [IntegrOmics](http://math.univ-toulouse.fr/biostat)                 | A tool for visualizing and analyzing omics data in the context of biological pathways and networks.                                                        | CCA                           | Genomics, Transcriptomics, Proteomics, Metabolomics | Unravels the relationship between two omics                                           | R                    | Integrates diverse omics data types and facilitates comprehensive analysis.                                                           | Complex data integration process, the potential for batch effects.                                                                     | Standardizing data formats, addressing platform-specific biases.            | Le Cao et al., 2009               |

### D. Bayesian method

| Tool                        | Description                                                                                                                                               | Algorithm                     | Type of Data                              | Use-case Addressed                                                            | Programming Language | Advantages                                                                                                                                 | Disadvantages                                                                                                                         | Challenges                                               | References                         | URL Link                                                    |
|-----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------|-------------------------------------------|--------------------------------------------------------------------------------|----------------------|-------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------|-----------------------------------|------------------------------------------------------------|
| [Multiple Dataset Integration (MDI)](https://github.com/smanson/mdiapp) | A method for analyzing multiple omics datasets to identify shared and unique features.                                                              | Correlated clustering          | Genomics, Transcriptomics, Proteomics, Metabolomics | Identifying shared and unique features across multiple omics datasets                        | R                    | Accounts for correlations across datasets identify shared and dataset-specific features.                                              | Computationally intensive, sensitive to prior assumptions.                                                                          | Specifying appropriate priors, and handling high-dimensional data.            | Kirk et al., 2012                  |
| TMD                        | A tool used for integrating and analyzing transcriptomic data across multiple studies or conditions.                                                       | Hierarchical Bayesian models   | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating and analyzing transcriptomic data across multiple studies                         | R                    | Accounts for study-specific effects and identify shared and unique features.                                                        | Computationally intensive, sensitive to prior assumptions.                                                                          | Specifying appropriate priors, and handling high-dimensional data.            | Savage et al., 2010                |
|[Patient-Specific Data Fusion (PSDF)](https://sites.google.com/site/patientspecificdatafusion/) | A method for integrating and analyzing patient-specific omics data to guide personalized medicine approaches.                                     | Dirichlet process mixture models | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating and analyzing patient-specific omics data for personalized medicine | MATLAB               | The probabilistic framework identifies patient-specific subtypes and prognoses.            | Computationally intensive, sensitive to prior assumptions.                                        | Specifying appropriate priors, and handling high-dimensional data. | Yuan et al., 2011            |
|[Joint BAYES](https://arxiv.org/abs/2406.00778)| A Bayesian approach for joint analysis and modeling of multi-omics data.                                                                           | Bayesian factor analysis         | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrative modeling and analysis of multi-omics data                  | R                    | The probabilistic framework handles missing data and uncertainty estimates.               | Computationally intensive, sensitive to prior assumptions.                                        | Specifying appropriate priors, and interpreting complex models.      | Ray et al., 2014              |
|[BCC](https://www.jmlr.org/papers/v14/klami13a.html)                         | A clustering method based on Bayesian inference for integrative analysis of omics data.                                                            | Bayesian modeling, consensus clustering | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrative clustering of multi-omics data to identify subgroups        | Proprietary          | The probabilistic framework accounts for uncertainties.                                       | Computationally intensive, sensitive to prior assumptions.                                        | Specifying appropriate priors, and handling high-dimensional data. | Lock and Dunson, 2013         |
|[iBAG](http://odin.mdacc.tmc.edu/~vbaladan/)                       | Integrative Bayesian Analysis of Genomic data, used for integrative analysis and modeling of genomic data.                                         | Hierarchical Bayesian modeling   | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating and modeling multiple genomic data sources                   | Python               | The probabilistic framework handles missing data and uncertainty estimates.               | Computationally intensive, sensitive to prior assumptions.                                        | Specifying appropriate priors, and interpreting complex models.      | Jennings et al., 2013         |
|[MOFA](https://biofam.github.io/MOFA2/)                         | A tool for integrative factor analysis of multi-omics data to identify latent factors and associations.                                             | Factor analysis                  | Genomics, Transcriptomics, Proteomics, Metabolomics | Identifying shared and dataset-specific sources of variation            | R                    | The probabilistic model handles missing data and provides uncertainty estimates.           | Computationally intensive for large datasets, sensitive to prior assumptions.               | Selecting appropriate priors, and interpreting latent factors.       | Argelaguet et al., 2018       |

### E. Fusion-based method

| Tool                        | Description                                                                                                                                               | Algorithm                     | Type of Data                              | Use-case Addressed                                                            | Programming Language | Advantages                                                                                                                                 | Disadvantages                                                                                                                         | Challenges                                               | References                         | URL Link                                                    |
|-----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------|-------------------------------------------|--------------------------------------------------------------------------------|----------------------|-------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------|-----------------------------------|------------------------------------------------------------|
|PDF | A method for integrating and analyzing patient-specific omics data to guide personalized medicine approaches.                                     | Dirichlet process mixture models | Genomics, Transcriptomics, Proteomics, Metabolomics | Integrating and analyzing patient-specific omics data for personalized medicine | MATLAB               | The probabilistic framework identifies patient-specific subtypes and prognoses.            | Computationally intensive, sensitive to prior assumptions.                                        | Specifying appropriate priors, and handling high-dimensional data. | Yuan et al., 2011            | NA                                                          |

## Alignment Tools

| S. NO | Tool Name                  | Application         | Description                                                                                                   | URL Link                                                     |
|-------|----------------------------|---------------------|---------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------|
| 1     | [Burrows Wheeler Aligner (BWA)](https://bio-bwa.sourceforge.net/) | Genomics           | BWA is a popular tool for mapping low-divergent sequences against a large reference genome efficiently.       |
| 2     | [Bowtie](https://bowtie-bio.sourceforge.net/index.shtml)                     | Genomics           | Bowtie is an ultrafast and memory-efficient tool for aligning sequencing reads to large genomes.              |
| 3     | [STAR](https://github.com/alexdobin/STAR)                       | Genomics           | STAR is a highly efficient RNA-seq aligner that can align RNA-seq reads to a reference genome.                |
| 4     | [BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi)                      | Genomics/Proteomics | BLAST is a widely used tool for comparing nucleotide or protein sequences to sequence databases.               |
| 5     | [HISAT2](https://daehwankimlab.github.io/hisat2/)                     | Genomics           | HISAT2 is a fast and sensitive spliced alignment tool for RNA-seq data, capable of aligning to large genomes. |
| 6     | [MAFFT](https://mafft.cbrc.jp/alignment/software/)                     | Genomics/Proteomics | MAFFT is a multiple sequence alignment tool commonly used for aligning DNA/protein sequences.                 |
| 7     | [Muscle](https://www.ebi.ac.uk/Tools/msa/muscle/?stype=protein)                     | Proteomics          | Muscle is a program for creating multiple sequence alignments of protein sequences, emphasizing accuracy.      |
| 8     | [ClustalW](https://www.genome.jp/tools-bin/clustalw)                   | Genomics/Proteomics | ClustalW is a widely used multiple sequence alignment program for DNA or protein sequences.                   |
| 9     | [Clustal Omega](https://www.ebi.ac.uk/Tools/msa/clustalo/)              | Genomics/Proteomics | Clustal Omega is a multiple sequence alignment program for DNA, RNA or protein.                               |

Sources:

Mansoor, S., Hamid, S., Tuan, T. T., Park, J.-E. & Chung, Y. S. Advance computational tools for multiomics data learning. Biotechnology Advances 77, 108447 (2024). <https://doi.org/https://doi.org/10.1016/j.biotechadv.2024.108447>



---------

<main id="main" style="display: flex; border: 1px; padding: 1px;">

<left style="flex: 0.3; padding-right: 1px;background-color:rgb(255, 255, 255); font-size:18pt" markdown="span">

![alt text](image-12.png)
![alt text](image-11.png)

</left>

<right id="col_right" style="flex: .7; padding-left: 1px;background-color:rgb(255, 255, 255);font-size:20pt" markdown="span">

- **Length bias**: Longer genes are more likely to generate more reads than shorter ones with similar expression levels.

- **Differences in library size composition**: Since the sequencing depth (number of reads per samples) is pre-determined as part of the sequencing strategy, skewed read count distribution within samples can mask finer details in gene expression profile.
  
- Between-sample differences in library size must be necessarily corrected before comparing samples to detect differentially expressed genes.

</right>
</main>

Francesca Finotello, Barbara Di Camillo, Measuring differential gene expression with RNA-seq: challenges and strategies for data analysis, Briefings in Functional Genomics, Volume 14, Issue 2, March 2015, Pages 130â€“142, <https://doi.org/10.1093/bfgp/elu035>

----
----

What biological questions can we answer?
 Usage limitations
----

# Supplemental Slides

---

## Read Count Normalization: What should I use?

- **Use RPKM/FPKM** when comparing gene expression within a **single sample** (e.g., understanding relative expression of genes within the same condition).
- **Use TPM** when comparing gene expression **between samples** because TPM ensures that total expression values are comparable across samples.
- **Use CPM** for **differential expression analysis**, especially with raw counts, as it retains variance information required by statistical methods (e.g., edgeR, DESeq2).

---